### DISCUSSION TITLE
Vision Portal getCameraState

### DISCUSSION THREAD

**Discussion by baqwas** [FIRST-Tech-Challenge/FtcRobotController#1328] [Q&A] [answered]
Hello!

I'm using a Mokose webcam at 640x480 with the sample **UtilityCameraFrameCapture** OpMode. I need help with two topics, please:
* _getCameraState_ returns the enum ERROR: is there any way to drill down into the reason?
* _saveNextFrameRaw_: is there some "completion code" to determine whether something was written?

FWIW, the webcam works: AprilTags are detected with the current SDK, and Live Stream is active.

I understand that this is an _unsupported webcam_, and I am **not seeking direct guidance on camera calibration**. I need to know if I am missing some documentation on VisionPortal methods or even some gray cells. Thanks.

Regards.

**Comment by baqwas** [comment]
**Update**:
_getCameraState_ is now returning STREAMING! Recycled power many, many times and checked Camera Stream! And now magically this state is **persistent** with no **known** change in code. Please disregard the first bullet point in the original inquiry. After thinking more about this item, ERROR should be fine for my use. Ideally, some generic error condition could be reported, but it would be impractical to report with granularity given the vast range of device error conditions where ERROR is probably is best lowest common denominator.

However, I still cannot determine where the captured image is saved on the desktop. Thanks again for any guidance.

Regards.

**Comment by WestsideRobotics** [comment, marked as answer]
> However, I still cannot determine where the captured image is saved on the desktop. 

The search box at **ftc-docs** shows two references to the sample `UtilityCameraFrameCapture`:

1. https://ftc-docs.firstinspires.org/en/latest/apriltag/vision_portal/apriltag_camera_calibration/apriltag-camera-calibration.html#utility-opmode

- images are stored on the RC device, not at the PC desktop
- shows the image filename format
- recommends transferring to desktop after each test run, to prevent over-writing

2. https://ftc-docs.firstinspires.org/en/latest/programming_resources/vision/camera_calibration/camera-calibration.html#calibrating-with-3df-zephyr

- gives the images' location on the RC device: root of USB device

**Reply by baqwas** [reply]
Thanks again, @WestsideRobotics!

My senility is on full display here. Now I remember being advised TWO years ago on this very topic.

Regards.
